plugins/common/tls/config.go

```diff
@@ -199,7 +199,6 @@ func loadCertificate(config *tls.Config, certFile, keyFile string) error {
 	}
 
 	config.Certificates = []tls.Certificate{cert}
-	config.BuildNameToCertificate()
```

Comment:
is removing this all together going to cause issues?

Reply:
`BuildNameToCertificate()` builds `c.NameToCertificate`

![image](https://user-images.githubusercontent.com/17336997/196656601-d8d44168-d0f3-409c-9499-550cb4c80c75.png)

It might be used in `getCertificate(clientHello *ClientHelloInfo) (*Certificate, error)` (when particular conditions are met)
![image](https://user-images.githubusercontent.com/17336997/196657263-9fc4253d-8644-47de-a09b-76247fcd74f6.png)

After removal of `config.BuildNameToCertificate()`, `c.NameToCertificate` will be `nil`, so this part of code might be responsible for choosing the best certificate for the given ClientHelloInfo instead:
```
	for _, cert := range c.Certificates {
		if err := clientHello.SupportsCertificate(&cert); err == nil {
			return &cert, nil
		}
	}
```

I'm also curious how this will work, but we should give it a try ;)

Reply:
great. Thanks @zak-pawel!